zerver/views/webhooks/gocd.py

```diff
@@ -0,0 +1,41 @@
+
+from __future__ import absolute_import
+from zerver.models import get_client
+from zerver.lib.actions import check_send_message
+from zerver.lib.response import json_success, json_error
+from zerver.decorator import REQ, has_request_variables, api_key_only_webhook_view
+import ujson
+
+@api_key_only_webhook_view
+@has_request_variables
+def api_gocd_webhook(request, user_profile, stream=REQ(default='gocd')):
+    try:
+        payload = ujson.loads(request.body)
+    except ValueError as e:
+        return json_error("Malformed JSON input, ", e.message)
+
+    try:
+        pipeline_name = payload["pipeline_name"]
+        pipeline_counter = payload["pipeline_counter"]
+        stage_name = payload["stage_name"]
+        stage_counter = payload["stage_counter"]
+        create_time = payload["create_time"]
+        last_transition_time = payload["last_transition_time"]
+        status = payload["status"]
+        server_url = payload["server_url"]
+        triggered_by = payload["triggered_by"]
+
+    except ValueError as e:
+        return json_error("Malformed JSON input, ", e.message)
```

Comment:
I think this should be checking for KeyError, since that's the exception you'd get in this section (you can test what exception it'd be by replacing e.g. "pipeline_name" with a field name that doesn't exist)